Core/Battler.gd

```diff
@@ -118,8 +126,32 @@ func get_all_targets(application_type : Enums.ApplicationType) -> Array[Entity]:
 			all_target = [PlayerManager.player]
 			
 	return all_target
+
   
-# TODO condition check for killing enemies and removing them from the combat
-# TODO condition check for killing player and ending the combat
-# TODO condition check for killing all enemies and ending the combat
+func _handle_enemy_deaths() -> void:
+	var enemies_to_remove : Array[Entity] = []
+	for enemy in _enemy_list:
+		if enemy.get_health_component().current_health == 0:
+			enemies_to_remove.append(enemy)
+			
+	for enemy in enemies_to_remove:
+		_enemy_list.erase(enemy)
+		enemy.queue_free()
+		
+	for enemy in _enemy_list:
+		enemy.get_party_component().set_party(_enemy_list)
+
+
+func _check_and_handle_battle_end() -> void:
+	if PlayerManager.player.get_health_component().current_health == 0:
+		PhaseManager.on_combat_end.emit(Enums.CombatResult.DEFEAT)
+	if _enemy_list.is_empty():
+		PhaseManager.on_combat_end.emit(Enums.CombatResult.VICTORY)
+
+
+func _handle_deaths() -> void:
+	_handle_enemy_deaths()
+	_check_and_handle_battle_end()
+
+
 # TODO reset temporary stats at the end of the combat using EntityStats.reset_modifier_dict_temp_to_default()
```

Comment:
this TODO should be done also

Reply:
Currently, I emit a signal and SceneController completely reloads the scene - this means entities get reloaded from 0 and their stats are gone. As such the todo doesn't really apply as the entity which holds the stats gets recreated per my understanding - which is why I left the todo in place so that it gets implemented once proper handling of combat progression is implemented

Reply:
The reset stat will reset the temp stats, which need to be reset anyway
Doing this would allow for proper testing of stat reset at combat end (though I already tested them in unit tests)

It also makes sense to leave it as is for when we have multiple combats I guess

Reply:
Per my understanding temp stats are still bound to the entity (so the player in our case I believe), which gets deleted and recreated since I reload the scene, or am I wrong there ? If that is the case, why is there a need to reset temp stats ?

Reply:
Yes it gets deleted for now, but later we will copy the stats, so we will need the reset stat function to be connected to the end of combat signal anyway

As I said, it's fine to leave it as is for now, since it is indeed related to stats carrying over, so not yet implemented